### PR TITLE
Declare term collection params so they can be exposed

### PR DESCRIPTION
**Pull request by danielbachhuber**
See #924

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@WP-API/amigos Can I get some validation on this concept? If it looks good, I'd like to merge, and then handle each controller separately.

**Comment by rmccue** [comment]
Looks fine by me.

File: lib/endpoints/class-wp-rest-controller.php [hunk 1]
```diff
@@ -173,6 +173,33 @@ public function get_item_schema() {
...
+	 * @return array
+	 */
+	public function get_collection_params() {
+		return array(
+			'page'                   => array(
+				'description'        => 'Current page of the collection.',
+				'type'               => 'integer',
+				'default'            => 1,
+				'sanitize_callback'  => 'absint',
+				),
```
Comment (rmccue) [inline lib/endpoints/class-wp-rest-controller.php]: Watch the indentation on your closing parens.

**Comment by danielbachhuber** [comment]
@joehoyle gave a [little pushback](https://wordpress.slack.com/archives/core-restapi/p1430867483005306) on this yesterday.

I still think it's a good idea for this metadata to be stored on the controller instead of global state, but it's worth discussing in tandem with #1047

**Comment by danielbachhuber** [comment]
> I still think it's a good idea for this metadata to be stored on the controller instead of global state

And, it makes subclassing possible.

**Pull request merged**